### ISSUE TITLE
Block Resets After Hibernation

### ISSUE THREAD

**Issue by marcodenic** [proginosko/LeechBlockNG-chrome#18]
when waking a machine from hibernate, LB block timers get reset despite not having visited the blocked site since block commenced.

**Comment by proginosko** [comment]
Please provide some specific steps to reproduce this issue. Also please specify your OS and browser (if not Google Chrome). Thanks!

**Comment by marcodenic** [comment]
I currently use a setup that allows access to a site 10 mins every hour.

open the site, let the timer expire so it's locked, now hibernate the machine. (note the expiry time - end of the current hour).

unhibernate the machine after the expiry time. click the link to return to the blocked page as we are now past the block time. timer resets and tells me to wait until the NEXT hour, despite the fact it could be 4 or 5 hours beyond the initial block period.

**Comment by proginosko** [comment]
I've been able to reproduce this (Chrome on Windows 10) but the hibernation seems to be irrelevant. The page remains blocked in the new time period even if the device is awake the whole time. Anyway, clearly this is a serious bug, so I will work on fixing it!

**Comment by marcodenic** [comment]
I must admit I noticed it happening a few times even without hibernate recently, so I was beginning to doubt hibernate had any effect.

glad you've tracked it down, thanks.

**Comment by proginosko** [comment]
This should be resolved in version 1.0.6.2. Reopen this issue if you discover otherwise.

**Comment by marcodenic** [comment]
Hi,

I have noticed it happening still, after resuming from hibernate.

time stands still while the machine is hibernated and the block never times out after resume.
I am using chromium edge.

reproduction details same as above, except I'm now using blur rather than the block page:

> I currently use a setup that allows access to a site 10 mins every hour.
> 
> open the site, let the timer expire so it's locked, now hibernate the machine. (note the expiry time - end of the current hour).
> 
> wake the machine after the expiry time. tab remains blurred despite being well passed what should have been the original block timeout.
> 

thanks.

**Comment by proginosko** [comment]
What happens if you refresh the blurred page (press F5)? Is it blurred when it's reloaded?

Which operating system are you using?

**Comment by marcodenic** [comment]
> What happens if you refresh the blurred page (press F5)? Is it blurred when it's reloaded?

it remains blurred. even if i disable the extension then re-enable it. refreshing ends up blurred again.

> 
> Which operating system are you using?

windows 10.0.19041 Build 19041
edge Version 88.0.673.0 (Official build) dev (64-bit)

**Comment by proginosko** [comment]
Interesting. I didn't realize you were using Edge. This could be an issue with Edge itself. But also, the Edge version of LB is only at version 1.0.6.2. I haven't yet updated it to the latest version (1.0.7.1). I'll do that soon, so let's see if that resolves it. If not, I'll do some testing on Edge itself.

**Comment by marcodenic** [comment]
yeah, I switched over from brave which I was using when I originally opened this issue.

**Comment by proginosko** [comment]
Okay, thanks. When you say hibernation, does that just mean closing your laptop? I want to make sure I'm reproducing the problem accurately.

**Comment by marcodenic** [comment]
well its a desktop machine, and its a timed hibernation set in power settings, but its basically the equivalent of hitting "hibernate" from the power dropdown in the start menu.

you may have to enable hibernate on your device, typically its sleep by default i believe:
https://windowsreport.com/hibernate-start-menu-windows-10/

**Comment by marcodenic** [comment]
still happening fyi.

just totally ignores the length of time the machine has been hibernating and pages remain locked as if no time has passed.
